src/test/java/at/ac/tuwien/kr/alpha/common/TermTest.java

```diff
@@ -66,4 +69,19 @@ public void testTermOrdering() throws Exception {
 		assertTrue(cto.compareTo(ft) < 0);
 		assertTrue(ft.compareTo(cto) > 0);
 	}
+
+	@Test
+	public void testStringVsConstantSymbolEquality() {
+		ConstantTerm<String> stringConstant = ConstantTerm.getInstance("string");
+		ConstantTerm<String> constantSymbol = ConstantTerm.getSymbolicInstance("string");
+		// reference equality must hold for both constructs..
+		Assert.assertTrue(stringConstant == ConstantTerm.getInstance("string"));
+		ConstantTerm<String> sameConstantSymbol = ConstantTerm.getSymbolicInstance("string");
+		Assert.assertTrue(constantSymbol == sameConstantSymbol);
+		// ... but stringConstant and constantSymbol are NOT the same thing!
+		Assert.assertNotEquals(stringConstant.hashCode(), constantSymbol.hashCode());
+		Assert.assertNotEquals(stringConstant, constantSymbol);
+		// ... same goes for compareTo, must behave in sync with equals and hashCode
+		Assert.assertNotEquals(0, stringConstant.compareTo(constantSymbol));
```

Comment:
Two minor things: First, I am not sure if we are guaranteed that the compiler will replace both `"string"` objects with exactly the same object; this could be enforced by making it an extra variable `String str = "string";` and then constructing both contant terms using `str`. Secondly, please adapt the comment style to the usual one, i.e., full sentences with upper-case first word.

Reply:
Both fixed. As far as I'm aware, string literals are compiled into being the same object, so as long as you don't create strings as `new String("stringLiteral")`, that should be fine - that being said, thanks for the heads-up, I changed the unit test to the bulletproof variant. Unfortunately I didn't find a good reference on Java's internal String handling in a quick google search.